code/modules/hydroponics/beekeeping/beehive.dm

```diff
@@ -1,308 +1,455 @@
-		return 1
+		var/transferred = min(G.reagents.maximum_volume - G.reagents.total_volume, reagents.total_volume)
+		reagents.trans_to_obj(I, transferred)
+		user.visible_message(SPAN_NOTICE("\The [user] collects honey from \the [src] into \the [G]."), SPAN_NOTICE("You collect [transferred] units of honey from \the [src] into \the [G]."))
+		return TRUE
+	return ..()
+
+/obj/machinery/honey_extractor/physical_attack_hand(mob/user)
+	if(inoperable())
+		return
+	if(time_done_processing > REALTIMEOFDAY)
+		to_chat(user, SPAN_WARNING("Wait until \the [src] stops processing first!"))
+		return
+	if(LAZYLEN(loaded_frames))
+		var/choice = show_radial_menu(user, src, list("turn on", "remove a frame", "remove all frames"))
+		switch(choice)
+			if("turn on")
+				START_PROCESSING_MACHINE(src, MACHINERY_PROCESS_SELF)
+				time_done_processing = REALTIMEOFDAY + 10 SECONDS
+				update_use_power(POWER_USE_ACTIVE)
+				update_icon()
+				return TRUE
+
+			if("remove a frame")
+				var/obj/item/honey_frame/H = loaded_frames[loaded_frames.len]
+				LAZYREMOVE(loaded_frames, H)
+				user.put_in_hands(H)
+				user.visible_message(SPAN_NOTICE("\The [user] take \a [H] from \the [src]."), SPAN_NOTICE("You take \a [H] from \the [src]."))
+				return TRUE
+
+			if("remove all frames")
+				user.visible_message(SPAN_NOTICE("\The [user] begins to remove the honey frames from \the [src]."), SPAN_NOTICE("You begin removing the honey frames from \the [src]."))
+				if(user.do_skilled(6 SECONDS, SKILL_BOTANY) && !(QDELETED(src) || QDELETED(user)) && LAZYLEN(loaded_frames))
+					for(var/obj/I in loaded_frames)
+						user.put_in_hands(I)
+					LAZYCLEARLIST(loaded_frames)
+					user.visible_message(SPAN_NOTICE("\The [user] removed all honey frames from \the [src]."), SPAN_NOTICE("You removed all honey frames from \the [src]."))
+					return TRUE
+				else
+					user.visible_message(SPAN_NOTICE("\The [user] stops removing honey frames from \the [src]."), SPAN_NOTICE("You stop removing honey frames from \the [src]."))
+	else
+		to_chat(user, SPAN_WARNING("There are no honey frames in \the [src]!"))
 
+/obj/machinery/honey_extractor/on_update_icon()
+	if(time_done_processing > REALTIMEOFDAY && operable())
+		icon_state = "centrifuge_moving"
+	else
+		icon_state = "centrifuge"
+
+/obj/machinery/honey_extractor/Process()
+	if(inoperable())
+		time_done_processing = 0
+		on_update_icon()
+		return PROCESS_KILL
+	if(REALTIMEOFDAY >= time_done_processing && LAZYLEN(loaded_frames))
+		extract_honey()
+		update_use_power(POWER_USE_IDLE)
+		return PROCESS_KILL
+	
+/obj/machinery/honey_extractor/proc/extract_honey()
+	if(!LAZYLEN(loaded_frames))
+		return
+	var/decl/material/M = GET_DECL(/decl/material/solid/wax/bees)
+	M.create_object(loc, LAZYLEN(loaded_frames), /obj/item/stack/material/puck)
+	for(var/obj/item/honey_frame/F in loaded_frames)
+		reagents.add_reagent(/decl/material/liquid/nutriment/honey, F.honey)
```

Comment:
You could possibly have a reagent reservoir in the comb that transfers into one in the extractor (beaker or such) rather than using a tracking var. Would allow for non-honey products too.

Reply:
Yeah, that makes sense! I just gotta wrap my head around the weird equation they made for honey production.